tests/end_to_end/test_sequenceAnnotations.py

```diff
@@ -0,0 +1,116 @@
+        frontend.reset()
+        frontend.configure(
+            baseConfig="DevelopmentConfig", extraConfig=config)
+        cls.app = frontend.app.test_client()
+
+    def sendSearchRequest(self, path, request, responseClass):
+        """
+        Sends the specified protocol request instance as JSON, and
+        parses the result into an instance of the specified response.
+        """
+        response = self.sendJsonPostRequest(path, request.toJsonString())
+        self.assertEqual(200, response.status_code)
+        responseData = responseClass.fromJsonString(response.data)
+        self.assertTrue(responseData.validate(responseData.toJsonDict()))
+        return responseData
+
+    def getAllDatasets(self):
+        path = 'datasets/search'
+        request = protocol.SearchDatasetsRequest()
+        responseData = self.sendSearchRequest(
+            path, request, protocol.SearchDatasetsResponse)
+        return responseData.datasets
+
+    def getAllFeatureSets(self):
+        datasetId = self.getAllDatasets()[0].id
+        path = 'featuresets/search'
+        request = protocol.SearchFeatureSetsRequest()
+        request.datasetId = datasetId
+        responseData = self.sendSearchRequest(
+            path, request, protocol.SearchFeatureSetsResponse)
+        return responseData.featureSets
+
+    def testSearchFeatures(self):
+        datasetId = self.getAllDatasets()[0].id
+        featureSets = self.getAllFeatureSets()
+        for featureSet in featureSets:
+            path = "features/search"
+            request = protocol.SearchFeaturesRequest()
+            request.datasetId = datasetId
+            request.featureSetId = featureSet.id
+            request.start = 0
+            request.end = 2**16
+            request.featureTypes = ["exon"]
+            request.referenceName = "chr1"
+            request.featureSetId = featureSet.id
+            responseData = self.sendSearchRequest(
+                path, request, protocol.SearchFeaturesResponse)
+            for feature in responseData.features:
+                print(feature.featureType)
+                self.assertIn(
+                    feature.featureType.term,
+                    request.featureTypes,
+                    "Term should be present {} {} {}".format(
+                        feature.featureType.term,
+                        request.featureTypes,
+                        feature))
+            path = "features/search"
+            request = protocol.SearchFeaturesRequest()
+            request.datasetId = datasetId
+            request.featureSetId = featureSet.id
+            request.start = 0
+            request.end = 2**16
+            request.featureTypes = ["gene", "exon"]
+            request.referenceName = "chr1"
+            request.featureSetId = featureSet.id
+            responseData = self.sendSearchRequest(
+                path, request, protocol.SearchFeaturesResponse)
+            for feature in responseData.features:
```

Comment:
the way the test is structured now, it could return zero results and still pass...